src/components/AddToWatchListButton/AddToWatchListButton.jsx

```diff
@@ -0,0 +1,93 @@
+
+import { useState } from 'react'
+import { useDispatch } from 'react-redux';
+import style from './AddToWatchListButton.module.css'
+import { toWatchList } from '../../store/filmApi/types';
+
+const buttonTypeArr = {
+    inWatch: {
+        type: 'inWatch',
+        url: './img/check-circle.svg',
+        class: 'inWatch',
+        tooltip: 'Remove from Watch list'
+    },
+    notInWatch: {
+        type: 'notInWatch',
+        url: './img/plus.svg',
+        class: 'notInWatch',
+        tooltip: 'Add to Watch list'
+    },
+    detailsInWatch: {
+        type: 'detailsInWatch',
+        text: <span>Delete from <br /> watch list</span>,
+        class: 'detailsInWatch'
+    },
+    detailsNotInWatch: {
+        type: 'detailsNotInWatch',
+        text: <span>Add to<br /> watch list</span>,
+        class: 'notInWatch'
+    }
+}
+
+const AddToWatchListButton = ({ name, image, genre, id, buttonType }) => {
+    const dispatch = useDispatch();
+    const [type, setType] = useState();
+    const [tooltipText, setTooltipText] = useState(buttonTypeArr.notInWatch.tooltip);
+    switch (type === undefined ? buttonType : type) {
```

Comment:
Very confusing using of switch. To be honest I can't understand anything :D Are you sure that there is not clearer way to do the job?)

Reply:
I have split one component to two different component for card and for details.